src/workspace/environment.rs

```diff
@@ -614,6 +623,55 @@ mod tests {
         );
     }
 
+     #[test]
+    fn test_activation_env_priority() {
+        let manifest = Workspace::from_str(
+            Path::new("pixi.toml"),
+            r#"
+            [project]
+            name = "foobar"
+            channels = []
+            platforms = ["linux-64", "osx-64"]
+
+            [activation.env]
+            FOO_VAR = "default"
+
+            [feature.foo.activation.env]
+            FOO_VAR = "foo"
+
+            [feature.cuda.activation.env]
+            FOO_VAR = "cuda"
+
+            [environments]
+            foo = ["foo"]
+            cuda = ["cuda"] 
```

Comment:
```suggestion
            cuda = ["foo", "cuda"] 
```

This way we ensure that "cuda" indeed has the highest priority